alpha/lib/model/KuserToUserRole.php

```diff
@@ -17,7 +17,45 @@ class KuserToUserRole extends BaseKuserToUserRole {
 
 	public function getCacheInvalidationKeys()
 	{
-		return array("kuserToUserRole:kuserId=".strtolower($this->getKuserId()));
+		return array("kuserToUserRole:kuserId=".strtolower($this->getKuserId()), "kuserToUserRole:id=".strtolower($this->getId()));
+	}
+	
+	public function setPuserId($puserId)
+	{
+		$kuser = kuserPeer::getKuserByPartnerAndUid(kCurrentContext::getCurrentPartnerId(), $puserId);
+		
+		if (!$kuser)
+		{
+			throw new kCoreException('[User ID Not Found]', kCoreException::INVALID_USER_ID, $puserId);
+		}
+		
+		parent::setKuserId($kuser->getId());
+	}
+	
+	public function getPuserId()
+	{
+		$kuserId = $this->getKuserId();
+		
+		// todo consult Yossi - when 'validateForUpdate' executes it calls the getter functions before we call setter functions
```

Comment:
Remove private comments if you you already handled them

Reply:
I did not, wanted your opinion
But I take if you are good with the solution ;)